tests/integration/worker/test_submission_worker.py

```diff
@@ -0,0 +1,394 @@
+        }
+        body = json.dumps(message)
+
+        mock_psm.side_effect = Exception("test error")
+
+        submission_worker.process_submission_callback(body)
+
+        mock_logger.assert_called_with("Exception while receiving message from submission queue with error test error")
+
+
+class ProcessSubmissionMessageTestClass(BaseTestClass):
+    @mock.patch("scripts.workers.submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
+    @mock.patch("scripts.workers.submission_worker.os.path.basename", return_value="user_annotation_file.txt")
+    @mock.patch("scripts.workers.submission_worker.run_submission")
+    def test_process_submission_message_succesfully(self, mock_rs, mock_basename):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": self.challenge_phase.pk,
+            "submission_pk": self.submission.id
+        }
+        user_annotation_file_path = "mocked/dir/submission_{}/user_annotation_file.txt".format(self.submission.id)
+
+        with mock.patch("scripts.workers.submission_worker.extract_submission_data") as mock_esd:
+            submission_worker.process_submission_message(message)
+            mock_esd.assert_called_with(self.submission.id)
+
+        submission_worker.process_submission_message(message)
+
+        mock_rs.assert_called_with(self.challenge.pk, self.challenge_phase, self.submission, user_annotation_file_path)
+
+    @mock.patch("scripts.workers.submission_worker.extract_submission_data")
+    def test_process_submission_message_when_submission_does_not_exist(self, mock_esd):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": self.challenge_phase.pk,
+            "submission_pk": self.submission.id
+        }
+        mock_esd.return_value = None
+
+        submission_worker.process_submission_message(message)
+
+        self.assertEqual(submission_worker.process_submission_message(message), None)
+
+    @mock.patch("scripts.workers.submission_worker.extract_submission_data")
+    @mock.patch("scripts.workers.submission_worker.logger.exception")
+    def test_process_submission_message_when_challenge_phase_does_not_exist(self, mock_logger, mock_esd):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": 1000,
+            "submission_pk": self.submission.id
+        }
+        mock_esd.return_value = self.submission
+        phase_id = 1000
+        with self.assertRaises(Exception):
+            submission_worker.process_submission_message(message)
+
+        mock_logger.assert_called_with("Challenge Phase {} does not exist".format(phase_id))
+
+
+class ExtractSubmissionDataTestClass(BaseTestClass):
+    @mock.patch("scripts.workers.submission_worker.logger.critical")
+    def test_extract_submission_data_when_submission_does_not_exist(self, mock_logger):
+        value = submission_worker.extract_submission_data(-1)
```

Comment:
Please don't hardcode the values. You can use something like `self.submission.pk + 100`